x/paramfilter/test/governance_params_test.go

```diff
@@ -0,0 +1,161 @@
+	suite.app.MintKeeper.SetMinter(suite.ctx, minter)
+}
+
+func TestHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(HandlerTestSuite))
+}
+
+func (suite *HandlerTestSuite) TestUnmodifiableParameters() {
+	testCases := []struct {
+		name     string
+		proposal *proposal.ParameterChangeProposal
+		onHandle func()
+		expErr   bool
+	}{
+		// The tests below show the parameters as modifiable, block in App.BlockedParams().
+		{
+			"bank.SendEnabled",
+			testProposal(proposal.ParamChange{
+				Subspace: banktypes.ModuleName,
+				Key:      string(banktypes.KeySendEnabled),
+				Value:    `[{"denom": "test", "enabled": false}]`,
+			}),
+			func() {
+				sendEnabledParams := suite.app.BankKeeper.GetParams(suite.ctx).SendEnabled
+				suite.Require().Equal([]*banktypes.SendEnabled{banktypes.NewSendEnabled("test", false)}, sendEnabledParams)
+			},
+			false,
+		},
+		{
+			"conensus.validator.PubKeyTypes",
+			testProposal(proposal.ParamChange{
+				Subspace: baseapp.Paramspace,
+				Key:      string(baseapp.ParamStoreKeyValidatorParams),
+				Value:    `{"pub_key_types": ["secp256k1"]}`,
+			}),
+			func() {
+				validatorParams := suite.app.BaseApp.GetConsensusParams(suite.ctx).Validator
+				suite.Require().Equal(
+					tmproto.ValidatorParams{
+						PubKeyTypes: []string{"secp256k1"},
+					},
+					*validatorParams)
+			},
+			false,
+		},
+		{
+			"consensus.Version.AppVersion",
+			testProposal(proposal.ParamChange{
+				Subspace: baseapp.Paramspace,
+				Key:      string(baseapp.ParamStoreKeyVersionParams),
+				Value:    `{"app_version": "3"}`,
+			}),
+			func() {
+				versionParams := suite.app.BaseApp.GetConsensusParams(suite.ctx).Version
+				suite.Require().Equal(
+					tmproto.VersionParams{
+						AppVersion: 3,
+					},
+					*versionParams)
+			},
+			false,
+		},
+		{
+			"staking.BondDenom",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyBondDenom),
+				Value:    `"test"`,
+			}),
+			func() {
+				stakingParams := suite.app.StakingKeeper.GetParams(suite.ctx)
+				suite.Require().Equal(
+					stakingParams.BondDenom,
+					"test")
+			},
+			false,
+		},
+		{
+			"staking.MaxValidators",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyMaxValidators),
+				Value:    `1`,
+			}),
+			func() {
+				stakingParams := suite.app.StakingKeeper.GetParams(suite.ctx)
+				suite.Require().Equal(
+					stakingParams.MaxValidators,
+					uint32(1))
+			},
+			false,
+		},
+		{
+			"staking.UnbondingTime",
+			testProposal(proposal.ParamChange{
+				Subspace: stakingtypes.ModuleName,
+				Key:      string(stakingtypes.KeyUnbondingTime),
+				Value:    `"1"`,
+			}),
+			func() {
+				stakingParams := suite.app.StakingKeeper.GetParams(suite.ctx)
+				suite.Require().Equal(
+					stakingParams.UnbondingTime,
+					time.Duration(1))
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
[question] is this idiom needed if this test isn't executed asynchronously? I've observed that this line is needed for concurrent code but isn't strictly necessary in non concurrent code.

```suggestion
```

Reply:
Thank you for spotting that! I have removed it now.